"""
these unicode symbols look good in the Godot font and work well with Label3D

https://symbl.cc/en/search/?q=Heart+symbols
https://www.unicode.org/emoji/charts/full-emoji-list.html

notable exceptions in Godot so far is the
gold coin: ğŸª™

also some look better in Godot
shield: ğŸ›¡

https://symbl.scc/en/search/?q=Heart+symbols
https://www.unicode.org/emoji/charts/full-emoji-list.html


seems to find related:
    https://emojiterra.com/spider-web/

"""
@tool
extends Node


func _ready():
    pass

func macro_update_label():
    $Label3D.text = symbols
    
func macro_iterating_chars_with_unicode_is_bad():
    var s = "ğŸ‘¨â€âœˆï¸ğŸ‘©â€âœˆï¸ğŸ‘¨â€ğŸš€ğŸ‘©â€ğŸš€" # 4 chars
    for char in s: # becomes different chars
        print(char)
    
func macro_test():
    
    var s = ""
    
    var count = 0
    
    var dict = {}
    for char in symbols:
        dict[char] = null
    
    var i = 0
    for key in dict:
        i += 1
        s += key
        if i % 10 == 0:
            s += "\n"
    $Label3D.text = s
    print(s)

var symbols = """

ğŸ”«ğŸš€ğŸ’£ğŸ’¥âš¡â¤ï¸ğŸ—ï¸ğŸ’ŠğŸ’‰â˜ ï¸ğŸ”¥ğŸ›¡ğŸ’²ğŸ’¯â˜¢ğŸ§¨
ğŸ‘‘ğŸ’°ğŸ’ğŸ’ğŸ†ğŸ
ğŸ¥‡ğŸ¥ˆğŸ¥‰
âœ¨ğŸ†ğŸ‡
ğŸ‚ğŸ°ğŸ¦

ğŸ’¬ğŸ’¤ğŸ’­âœ‰ï¸ğŸŒğŸŒğŸš¬ğŸ·ğŸŒ¼ğŸ›¸ğŸªâ˜€ï¸â˜ï¸

ğŸ’©ğŸ¤¡ğŸ‘½ğŸ¤–ğŸ‘¹ğŸ‘ºğŸ‘¶ğŸ§’ğŸ‘¦ğŸ‘§ğŸ§‘ğŸ‘±ğŸ‘¨ğŸ§”ğŸ‘¨â€ğŸ¦°ğŸ‘¨â€ğŸ¦±ğŸ‘¨â€ğŸ¦³ğŸ‘¨â€ğŸ¦²ğŸ‘©ğŸ‘©â€ğŸ¦°ğŸ‘©â€ğŸ¦±ğŸ‘©â€ğŸ¦³ğŸ‘©â€ğŸ¦²ğŸ‘±â€â™€ï¸ğŸ‘±â€â™‚ï¸ğŸ§“ğŸ‘´ğŸ‘µğŸ¤´ğŸ‘¸ğŸ‘³ğŸ‘³â€â™€ï¸ğŸ‘²
ğŸ‘¼ğŸ…ğŸ¤¶ğŸµğŸ¶ğŸ¦ŠğŸ¦ğŸ±ğŸ˜ºğŸ¦ğŸ¯ğŸ·ğŸ—ğŸ¹ğŸ»ğŸ¨ğŸ¼ğŸ¸

ğŸ°
ğŸ’ğŸ“ğŸ‹ğŸŠğŸ’ğŸ‡ğŸ””ğŸ‰7
ğŸŒğŸğŸŠğŸ¹
ğŸğŸ
ğŸ¥ğŸ”
ğŸ§ªğŸ¦ ğŸ’€
â™ ï¸â™¥ï¸ï¸â™£â™¦ï¸â™¡
â­
ğŸ’”ğŸ’™ğŸ’›ğŸ’šğŸ’œğŸ–¤ğŸ¤
ğŸ”‘ğŸšª
â˜˜ï¸ğŸŒˆğŸŒ·
ğŸ¦‡ğŸŒ™ğŸŒ•ğŸ¦‰ğŸ•·ï¸ğŸ§›â€â™‚ï¸ğŸ•¸ï¸ğŸƒğŸ‘»âš°ï¸ğŸ©¸
ğŸ¥šğŸ£ğŸ¥ğŸ¤ğŸ“
ğŸ—
ğŸ¦˜ğŸ¦ğŸ§ğŸ§™ğŸ²ğŸ”ğŸ£ğŸ­
ğŸŒŸğŸŒ ğŸŒˆâ˜£
ğŸ’€ğŸ‘»ğŸ‘¾ğŸ’«ğŸ§¸

ğŸ‘¨â€âš•ï¸ğŸ‘©â€âš•ï¸ğŸ‘¨â€ğŸ“ğŸ‘©â€ğŸ“ğŸ‘¨â€ğŸ«ğŸ‘©â€ğŸ«ğŸ‘¨â€âš–ï¸ğŸ‘©â€âš–ï¸
ğŸ‘¨â€ğŸŒ¾ğŸ‘©â€ğŸŒ¾ğŸ‘¨â€ğŸ³ğŸ‘©â€ğŸ³ğŸ‘¨â€ğŸ”§ğŸ‘©â€ğŸ”§ğŸ‘¨â€ğŸ­ğŸ‘©â€ğŸ­ğŸ‘¨â€ğŸ’¼ğŸ‘©â€ğŸ’¼ğŸ‘¨â€ğŸ”¬ğŸ‘©â€ğŸ”¬ğŸ‘¨â€ğŸ¤ğŸ‘©â€ğŸ¤ğŸ‘¨â€ğŸ¨ğŸ‘©â€ğŸ¨ğŸ‘¨â€âœˆï¸ğŸ‘©â€âœˆï¸ğŸ‘¨â€ğŸš€ğŸ‘©â€ğŸš€
"""


